src/map/instance.cpp

```diff
@@ -708,6 +708,25 @@ int instance_addmap(int instance_id) {
 	return idata->map.size();
 }
 
+/**
+ * Fills outname with the name of the instance map name
+ * @param map_id: Mapid to use
+ * @param instance_id: Instance id
+ * @param outname: Pointer to allocated memory that will be filled in
+ */
+void instance_generate_mapname(int map_id, int instance_id, char outname[MAP_NAME_LENGTH]) {
+
+#if MAX_MAP_PER_SERVER > 9999
+	#error This algorithm is only safe for up to 9999 maps, change at your own risk.
+#endif
+	// Since we can't truncate integers like we can strings, allocate enough of a buffer
+	// to hold two integers, a symbol, and \0, then copy it to outname
+	char name[30];
```

Comment:
We truncate integers with our algorithm, so you can go straight for the outname pointer.